### ISSUE TITLE
Proposal: Change how "rerun" works in 2.0

### ISSUE THREAD

**Issue by krancour** [brigadecore/brigade#1062]
Currently, `brig rerun`, according to interactive help behaves as follows:

> Request that Brigade re-run a build.
>
> This will clone an old event, assign it a new build ID, and resubmit it. The build
must still be accessible. Once a build is vacuumed, it can no longer be re-run.
>
> Using the '-f' flag will cause brig to resend the old payload, but override the old
script with the provided one.
>
> As this command features all of the same flags as run, additional overrides are available,
such as re-running with a different event, payload body, commit, etc.  See usage via -h/--help.

The ability to change the payload, the script that executes, etc., to me, feels like this doesn't _necessarily_ effect a _re_-run, but effectively creates a whole new event.

I'd like to propose that in 2.0, "rerun" clones and "retries" an existing event... full stop. If you want to do something different than what was done the first time around, `brig run` should be used to create a new event.

imho, this newly proposed behavior tracks better with what I presume is most people's general understanding of the verb "rerun."

**Comment by radu-matei** [comment]
But would this effectively make things more difficult for people actively trying to change some parameters of a previous event execution?

**Comment by krancour** [comment]
Yes. But is that "rerunning?" Or is it something different? I'm proposing this mainly on the grounds that the current behavior doesn't comport with the common understanding of the word "rerun"-- to run again.

I could imagine a new option on `run` to utilize an existing build as a template.

**Comment by krancour** [comment]
I believe we've put closure on this now, so I'm tentatively closing it and invite it to be re-opened if anyone deems it necessary.

Where we landed on this in v2 is that "rerun" go decomposed into two separate commands:

1. `clone` makes a copy of an existing event-- same source, type, payload, and git details (if applicable)-- and does not afford any opportunity to change event details
2. `retry` (wip) is similar to clone, but inherits the status of any successful job related to the original event

Nowhere in here is any ability to "clone + modify." I had objected to that on security grounds because is it really a good idea to allow a user to create what is really a net new event with arbitrary details that _appear_ to have originated from some gateway but didn't?

(Aside, following the same reasoning, we might consider removing some options from initial event creation via CLI also. Should source really be user-specifiable?)

@vdice proposed that "clone + modify" was useful for troubleshooting purposes-- and I agree. _But_ a new feature of v2 is that a Brigadier-based `brigade.js` script can be executed (smoke tested) locally, with dummy events that can be user specified. i.e. The ability exists to provide a dummy event and see what Brigade _would_ do with that event without actually involving Brigade itself. In light of this, the troubleshooting story has probably improved without the need to provide any "clone + modify" capability.
